osu.Game/Beatmaps/Drawables/DifficultySpectrumDisplay.cs

```diff
@@ -159,21 +160,25 @@ private void updateDisplay()
                 countText.Alpha = collapsed ? 1 : 0;
                 countText.Text = beatmaps.Length.ToLocalisableString(@"N0");
 
+                var visibleBeatmaps = beatmaps
+                                      .Where(b => !(b is BeatmapInfo bi && bi.Hidden))
+                                      .ToArray();
+
                 var dots = this.OfType<DifficultyDot>().ToArray();
+                int count = Math.Min(visibleBeatmaps.Length, dots.Length);
 
-                for (int i = 0; i < max_difficulties_before_collapsing; i++)
+                for (int i = 0; i < count; i++)
```

Comment:
I'm 99% sure this is wrong. It's no longer iterating all dots, which means some may be left in the wrong state on pooling reuse.